packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -58,9 +58,19 @@ async function initState(
     docsService.update(doc.uri, { cwd, title: cwd });
   };
 
+  const removeInitCommand = () => {
+    if (doc.kind !== 'doc.terminal_shell') {
+      return;
+    }
+    // The initCommand has to be launched only once, not every time we recreate the document from
+    // the state.
+    docsService.update(doc.uri, { initCommand: undefined });
+  };
+
```

Comment:
It seems like a hack to me, but given the other functions in this hook I think we're fine for now. Once we add more stuff here it should probably get reorganized.

Reply:
Also, I understand now why you defined that one function inside a component or hook rather than outside. 🙊

Reply:
Ah, forgot to add that since launching a command means just adding a new document and then the side effect happen when the document gets rendered, this seems like the only way in which we can implement "Launch this command, but only once and not each time this doc is recreated".

I guess we could also have a boolean like `hasInitCommandBeenLaunched`, but for now we wouldn't even need that.

Reply:
Yeah, it looks like a hack, but it's fine for now